### PR TITLE
Add support for hdf5 files

### PR DESCRIPTION
**Pull request by balbok0**
* Adds support for HDF5 files. Right now nothing fancy, just reads all of the data and transforms it into concatenated NumPy arrays.

I also added conda environment file for easier setup for other folks

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -21,21 +22,7 @@
 
 
     #np.set_printoptions(precision=3, suppress=True)
-
```
Comment (nhatch) [inline train.py:24]: Good idea pulling this into a separate file

File: load_data.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            data = row[1:].reshape((1,-1))
+            if row[0] != seq_no:
+                if len(seq) > 1:
+                    seqs.append(seq)
+                seq = data
+                seq_no = row[0]
+            else:
+                seq = np.concatenate([seq, data], 0)
+        return seqs
+    except OSError as e:
```
Comment (nhatch) [inline load_data.py:81]: IMO it is inadvisable to use exceptions for normal control flow situations. I would prefer for us to check whether these files exist using something like this https://www.pythontutorial.net/python-basics/python-check-if-file-exists/

File: models.py [hunk 3]
```diff
@@ -66,7 +67,10 @@ def train(self, seqs, n_steps=1):
         weighted_y = train_y * WEIGHTS
         assert(weighted_y.shape == train_y.shape)
         assert(len(train_x) == len(train_y))
-        weighted_w = np.linalg.solve(train_x.T @ train_x, train_x.T @ weighted_y)
+        weighted_w = np.linalg.solve(
+            train_x.T @ train_x + self.lambda_ * np.eye(train_x.shape[1]),
+            train_x.T @ weighted_y,
+        )
```
Comment (nhatch) [inline models.py:73]: Hmm, you didn't mention this change in the PR description! Can't hurt to add support for ridge regression though; this is good

File: train.py [hunk 4]
```diff
@@ -106,15 +98,16 @@
...
+        uni_model = UnicycleModel(D=D, H=H, P=P, delay_steps=1)
+        uni_model.train(train_set, n_steps=N_TRAIN_STEPS)
+        _, _, umx, umy = uni_model.compare_qualitative(
+                viz_seq, start_idx=viz_start_idx, n_steps=n_steps)
+        line, = plt.plot(umx, umy, color='red', label='Unicycle model')
+        uni_score = uni_model.evaluate(test_set, n_steps=N_EVAL_STEPS)
+        print("Unicycle model:", uni_score)
 
-    linear_model = LinearModel(D=D, H=H, P=P, delay_steps=1)
+    linear_model = LinearModel(D=D, H=H, P=P, delay_steps=1, lambda_=1e-2)
```
Comment (nhatch) [inline train.py:110]: How did you choose this value for lambda? Did this give better results for something?

File: load_data.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        for topic in topics:
+            topic_data = h5_file[topic][:]
+            sim_data_raw.append(topic_data)
+        sim_data_raw = np.concatenate(sim_data_raw, axis=1)
+
+        # H
+        # linear.x
+        sim_data_raw[:, len(topics) + 1] = h5_file["input_pair/state/twist/linear"][:, 0]
+        # angular.z
+        sim_data_raw[:, len(topics) + 1] = h5_file["input_pair/state/twist/angular"][:, 2]
```
Comment (nhatch) [inline load_data.py:29]: It looks like this line might overwrite the previous line?

File: load_data.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+        # P
+
+        # Planer
+
+        print("State")
```
Comment (nhatch) [inline load_data.py:35]: I'm confused by these comments and this print statement

File: load_data.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    if "rzr" in dataset_name:
+        # 1. Extract data
+        # D
+        topics = [
+            # D
+            "input_pair/control/steer", "input_pair/control/throttle", "input_pair/control/steer",
+            # H
+            "input_pair/state/twist/linear", "input_pair/state/twist/angular",
+            # P
+            "input_pair/state/pose/position", "input_pair/state/pose/orientation", "input_pair/state/twist/linear", "input_pair/state/twist/angular",
```
Comment (nhatch) [inline load_data.py:16]: So this is my fault for not having documented / written this code very well, but: `H` is supposed to be odom information including both pose and twist. `P` is the _ground truth_ pose and twist information. Does your H5 file include ground truth pose information? For the datasets which are already in this repo, I collected that from the simulator using the `/unity_command/ground_truth/...` topics.

**Review by nhatch — COMMENTED** [review]
Sorry for the delay reviewing this! Hopefully now that the demo's done we can focus more on dynamics modeling, though I might personally still be distracted by GPS stuff for another week or two....

**Comment by balbok0** [comment]
Closing this PR in favor of #2

**Pull request closed without merging**